artichoke-backend/src/extn/core/integer/mod.rs

```diff
@@ -88,6 +89,36 @@ impl Integer {
         }
     }
 
+    unsafe extern "C" fn bit(mrb: *mut sys::mrb_state, slf: sys::mrb_value) -> sys::mrb_value {
+        fn method(interp: &Artichoke, value: Value, other: Value) -> Result<Value, ArtichokeError> {
+            let value = value.try_into::<Int>()?;
```

Comment:
usually these two conversions are differentiated. This first one is a failure to turn a receiver we know is an `Integer` (because methods are being dispatched on it) into an `Int`. Code like this usually propagates this as a fatal exception:

https://github.com/artichoke/artichoke/blob/e38ad36811360623d69a350a587e6b9886d2a14b/artichoke-backend/src/extn/core/array/trampoline.rs#L12-L17